CalculatedVariables/src/CalculatedVariablesEngine.cpp

```diff
@@ -87,6 +87,33 @@ double* CalculatedVariables::Engine::parserVariableRequestHandler(const char* na
     }
 }
 
+static std::string elaborateParent(
+        const std::string& input,
+        unsigned int levels,
+        const std::string& thisFormulaAddress)
+{
+    // the assumption is that for every level to go up we find the dot and cut off at such place
+    LOG(Log::TRC, logComponentId) << "Elaborating parent of address='" << input << "' for numLevels='" << levels << "'";
+    std::size_t cutOffIndex = std::string::npos;
+    while (levels > 0)
+    {
+        cutOffIndex = input.rfind('.', cutOffIndex);
+        if (cutOffIndex == std::string::npos)
+            LOG_AND_THROW_ERROR(thisFormulaAddress, "Not enough levels to go up!"); 
+        levels--;
+        if (levels>0)
+        {
+            if (cutOffIndex>0)
+                cutOffIndex--;
+            else
```

Comment:
It seems that lines 101-102 already cover you from this else condition to be happening.